File: Classes/Structure/Store/bin.vb
''' <summary>
''' DataType representing 1 bin with 1 article through string().
''' </summary>
Public Class bin
	Private stringBinPath As String = "Nothing"
	Private stringBinFile As String = "Nothing"
	Private stringLinkPath As String = "Nothing"
	Private stringLinkFile As String = "Nothing"
	Private integerLinkStorage As Integer = 0
	Private integerLinkSection As Integer = 0
	Private integerLinkArticle As Integer = 0
	Private integerBinQuantity As Integer = 0
	
	Private fileHandler As New binaryFileHandler
	Private stringBinData() As String
''' <summary>
''' Resizes stringDataBin to 3 (LinkStorage, LinkSection, LinkArticle, Quantity).
''' </summary>
	Public Sub New()
		ReDim stringBinData(3)
	End Sub
''' <summary>
''' Loads Bin and Item Data.
''' </summary>
''' <param name="PathShelf">Needed for path and file seperation.</param>
	Public Sub BinLoad(PathShelf As String)
		stringBinData = fileHandler.LoadRow(PathShelf, stringBinFile)
		integerLinkStorage = CType(stringBinData(0), Integer)
		integerLinkSection = CType(stringBinData(1), Integer)
		integerLinkArticle = CType(stringBinData(2), Integer)
		integerBinQuantity = CType(stringBinData(3), Integer)
		
		stringLinkPath = gamecache.playerStorage.arraySection(integerLinkSection).arrayArticle(integerLinkArticle).ArticlePath
		stringLinkFile = gamecache.playerStorage.arraySection(integerLinkSection).arrayArticle(integerLinkArticle).ArticleFile
	End Sub
''' <summary>
''' Saves the Bin and loads the linked Item Data.
''' </summary>
''' <param name="PathShelf">Needed for path and file seperation.</param>
	Public Sub BinSave(PathShelf As String)
		stringBinData(0) = CType(integerLinkStorage, String)
		stringBinData(1) = CType(integerLinkSection, String)
		stringBinData(2) = CType(integerLinkArticle, String)
		stringBinData(3) = CType(integerBinQuantity, String)
		fileHandler.Save(PathShelf, stringBinFile, , stringBinData)
		
		stringLinkPath = gamecache.playerStorage.arraySection(integerLinkSection).arrayArticle(integerLinkArticle).ArticlePath
		stringLinkFile = gamecache.playerStorage.arraySection(integerLinkSection).arrayArticle(integerLinkArticle).ArticleFile
	End Sub
''' <summary>
''' Get/Adjust the path of the Bin without ItemLoad().
''' </summary>
	Public Property BinPath As String
		Get
			Return stringBinPath
		End Get
		Set (Value As String)
			stringBinPath = Value
		End Set
	End Property
''' <summary>
''' Get/Adjust the filename of the Bin without ItemLoad().
''' </summary>
	Public Property BinFile As String
		Get
			Return stringBinFile
		End Get
		Set (Value As String)
			stringBinFile = Value
		End Set
	End Property
''' <summary>
''' Get/Adjust the Storage linked for this Bin.
''' </summary>
	Public Property LinkStorage As Integer
		Get
			Return integerLinkStorage
		End Get
		Set (Value As Integer)
			integerLinkStorage = Value
		End Set
	End Property
''' <summary>
''' Get/Adjust the Section linked for this Bin.
''' </summary>
	Public Property LinkSection As Integer
		Get
			Return integerLinkSection
		End Get
		Set (Value As Integer)
			integerLinkSection = Value
		End Set
	End Property
''' <summary>
''' Get/Adjust the Article linked for this Bin.
''' </summary>
	Public Property LinkArticle As Integer
		Get
			Return integerLinkArticle
		End Get
		Set (Value As Integer)
			integerLinkArticle = Value
		End Set
	End Property
''' <summary>
''' Get/Adjust the Quantity without BinSave().
''' </summary>
	Public Property BinQuantity As Integer
		Get
			Return integerBinQuantity
		End Get
		Set (Value As Integer)
			integerBinQuantity = Value
		End Set
	End Property
	''' <summary>
	''' Get the Item Name from Linked Article.
	''' </summary>
	ReadOnly Public Property ArticleName As String
		Get
			Return gamecache.playerStorage.arraySection(integerLinkSection).arrayArticle(integerLinkArticle).ItemLink.Name_Title
		End Get
	End Property
	''' <summary>
	''' Get the Last Selling price from Linked Article.
	''' </summary>
	ReadOnly Public Property ArticleLastSell As Integer
		Get
			Return gamecache.playerStorage.arraySection(integerLinkSection).arrayArticle(integerLinkArticle).LastSell
		End Get
	End Property
End Class

File: Game/Classes/Structure/Store/bin.cs
using System;

/// <summary>
/// DataType representing 1 bin with 1 article through string().
/// </summary>
public class bin
{
    private string stringBinPath = "Nothing";
    private string stringBinFile = "Nothing";
    private string stringLinkPath = "Nothing";
    private string stringLinkFile = "Nothing";
    private int integerLinkStorage = 0;
    private int integerLinkSection = 0;
    private int integerLinkArticle = 0;
    private int integerBinQuantity = 0;

    private binaryFileHandler fileHandler = new binaryFileHandler();
    private string[] stringBinData;

    /// <summary>
    /// Resizes stringDataBin to 3 (LinkStorage, LinkSection, LinkArticle, Quantity).
    /// </summary>
    public bin()
    {
        stringBinData = new string[4];
    }

    /// <summary>
    /// Loads Bin and Item Data.
    /// </summary>
    /// <param name="PathShelf">Needed for path and file seperation.</param>
    public void BinLoad(string PathShelf)
    {
        string[] stringMediator = null;
        stringMediator = stringBinPath.Split(Char(92));
        stringBinFile = stringMediator(stringMediator.GetUpperBound(0));
        stringBinPath = stringBinPath.Remove(stringBinPath.Length - stringBinFile.Length, stringBinFile.Length);
        stringBinData = fileHandler.LoadRow(PathShelf, stringBinFile);
        integerLinkStorage = Convert.ToInt32(stringBinData[0]);
        integerLinkSection = Convert.ToInt32(stringBinData[1]);
        integerLinkArticle = Convert.ToInt32(stringBinData[2]);
        integerBinQuantity = Convert.ToInt32(stringBinData[3]);

        stringLinkPath = gamecache.currentCharacterStorage.arraySection[integerLinkSection].arrayArticle[integerLinkArticle].ArticlePath;
        stringLinkFile = gamecache.currentCharacterStorage.arraySection[integerLinkSection].arrayArticle[integerLinkArticle].ArticleFile;
    }
    
    /// <summary>
    /// Saves the Bin and loads the linked Item Data.
    /// </summary>
    /// <param name="PathShelf">Needed for path and file seperation.</param>
    public void BinSave(string PathShelf)
    {
        stringBinData[0] = Convert.ToString(integerLinkStorage);
        stringBinData[1] = Convert.ToString(integerLinkSection);
        stringBinData[2] = Convert.ToString(integerLinkArticle);
        stringBinData[3] = Convert.ToString(integerBinQuantity);
        fileHandler.Save(PathShelf, stringBinFile, , stringBinData);

        stringLinkPath = gamecache.currentCharacterStorage.arraySection[integerLinkSection].arrayArticle[integerLinkArticle].ArticlePath;
        stringLinkFile = gamecache.currentCharacterStorage.arraySection[integerLinkSection].arrayArticle[integerLinkArticle].ArticleFile;
    }
    
    /// <summary>
    /// Get/Adjust the path of the Bin without ItemLoad().
    /// </summary>
    public string BinPath {
        get { return stringBinPath; }
        set { stringBinPath = value; }
    }
    
    /// <summary>
    /// Get/Adjust the filename of the Bin without ItemLoad().
    /// </summary>
    public string BinFile {
        get { return stringBinFile; }
        set { stringBinFile = value; }
    }
    
    /// <summary>
    /// Get/Adjust the Storage linked for this Bin.
    /// </summary>
    public int LinkStorage {
        get { return integerLinkStorage; }
        set { integerLinkStorage = value; }
    }
    
    /// <summary>
    /// Get/Adjust the Section linked for this Bin.
    /// </summary>
    public int LinkSection {
        get { return integerLinkSection; }
        set { integerLinkSection = value; }
    }
    
    /// <summary>
    /// Get/Adjust the Article linked for this Bin.
    /// </summary>
    public int LinkArticle {
        get { return integerLinkArticle; }
        set { integerLinkArticle = value; }
    }
    
    /// <summary>
    /// Get/Adjust the Quantity without BinSave().
    /// </summary>
    public int BinQuantity {
        get { return integerBinQuantity; }
        set { integerBinQuantity = value; }
    }
    
    /// <summary>
    /// Get the Item Name from Linked Article.
    /// </summary>
    public string ArticleName {
        get { return gamecache.currentCharacterStorage.arraySection[integerLinkSection].arrayArticle[integerLinkArticle].ItemLink.Name_Title; }
    }
    
    /// <summary>
    /// Get the Last Selling price from Linked Article.
    /// </summary>
    public int ArticleLastSell {
        get { return gamecache.currentCharacterStorage.arraySection[integerLinkSection].arrayArticle[integerLinkArticle].LastSell; }
    }
}